pyaedt/icepak.py

```diff
@@ -3772,3 +3776,240 @@ def assign_source(
+    ):
+        """
+        Assign block boundary for solid objects.
+
+        Parameters
+        ----------
+        object_name : str
+            Name of the object.
+        power_assignment : str or dict
+            String with value and units of the power assignment or with ``"Joule Heating"``.
+            Also, a dictionary can be used for temperature dependent or transient assignment.
+            The dictionary should contain three keys: ``"Type"``, ``"Function"`` and
+            ``"Values"``. Accepted ``"Type"`` values are: ``"Temp Dep"`` and ``"Transient"``.
+            Accepted ``"Function"`` are: ``"Linear"``, ``"Power Law"``, ``"Exponential"``,
+            ``"Sinusoidal"``, ``"Square Wave"`` and ``"Piecewise Linear"``. ``"Temp Dep"`` only
+            support the latter. ``"Values"`` contains a list of strings containing the parameters
+            required by the ``"Function"`` selection (e.g. ``"Linear"`` requires two parameters:
+            the value of the variable at t=0 and the slope of the line). The parameters required by
+            each ``Function`` option is in Icepak documentation. The parameters must contain the
+            units where needed.
+        boundary_name : str, optional
+            Name of the source boundary. The default is ``None`` and the boundary name will be
+            generated automatically.
+        htc : float, str or dict, optional
+            String with value and units of heat transfer coefficient for the external conditions.
+            Also, a dictionary can be used for temperature dependent or transient assignment as
+            described in ``power_assignment``. The default is ``None``
+        ext_temperature : float, str or dict, optional
+            String with value and units of temperature for the external conditions.
+            Also, a dictionary can be used for transient assignment as described in ``power_assignment``.
+            The default is ``"AmbientTemp"`` and will have effect if ``htc`` is not ``None``
+
+
+        Returns
+        -------
+        :class:`pyaedt.modules.Boundary.BoundaryObject`
+            Boundary object when successful or ``None`` when failed.
+
+        References
+        ----------
+
+        >>> oModule.AssignBlockBoundary
+
+        Examples
+        --------
+        >>> from pyaedt import Icepak
+        >>> ipk = Icepak()
+        >>> ipk.solution_type = "Transient"
+        >>> box = ipk.modeler.create_box([5, 5, 5], [1, 2, 3], "BlockBox3", "copper")
+        >>> power_dict = {"Type": "Transient", "Function": "Sinusoidal", "Values": ["0W", 1, 1, "1s"]}
+        >>> block = ipk.assign_solid_block("BlockBox3", power_dict)
+        """
+        if not self.modeler.get_object_from_name(object_name).solve_inside:
+            self.logger.add_error_message(
+                "Please use ``assign_hollow_block`` function with this object as" "``solve_inside`` is ``False``."
```

Comment:
Avoid "Please" in technical documentation. Please omit in all error messages and adjust any tests.